analyzer/repo-cloner-plugin/src/main/java/eu/fasten/analyzer/repoclonerplugin/RepoClonerPlugin.java

```diff
@@ -116,37 +116,94 @@ public void consume(String record) {
             String product = null;
             if (artifact != null && !artifact.isEmpty()
                     && group != null && !group.isEmpty()) {
-                product = group + ":" + artifact + ((version == null) ? "" : ":" + version);
+                product = group + ":" + artifact + (version.isEmpty() ? "" : ":" + version);
             }
             outputPath = File.separator
                     + (artifact == null ? "" : artifact.charAt(0) + File.separator)
                     + artifact + File.separator
                     + (product == null ? "unknown" : product.replace(":", "_")) + ".json";
-            var repoUrl = json.optString("repoUrl");
-            if (repoUrl != null && !repoUrl.isEmpty()) {
-                try {
-                    var gitCloner = new GitCloner(baseDir);
-                    cloneRepo(repoUrl, gitCloner);
-                } catch (GitAPIException | IOException e) {
-                    logger.error("Error cloning repository"
-                            + (product == null ? "" : "for '" + product + "'")
-                            + " from " + repoUrl, e);
-                    this.pluginError = e;
-                    return;
+            var repoUrl = json.optString("repoUrl").replaceAll("[\\n\\t ]", "");;
+            if (!repoUrl.isEmpty()) {
+                var gitCloner = new GitCloner(baseDir);
+                var hgCloner = new HgCloner(baseDir);
+                var svnCloner = new SvnCloner(baseDir);
+                var cloned = cloneRepo(repoUrl, artifact, group, gitCloner, hgCloner, svnCloner);
+                if (!cloned) {
+                    logger.info("Could not clone repository of " + product + " from " + repoUrl);
+                    pluginError = new RuntimeException("Failed to clone repository of "
+                            + product + " from " + repoUrl);
                 }
                 if (getPluginError() == null) {
                     logger.info("Cloned repository"
-                            + (product == null ? "" : "of '" + product + "'")
+                            + (product == null ? "" : " of '" + product + "'")
                             + " from " + repoUrl + " to " + repoPath);
                 }
             } else {
                 logger.info("Repository URL not found");
             }
         }
 
-        public void cloneRepo(String repoUrl, GitCloner gitCloner)
-                throws GitAPIException, IOException {
-            repoPath = gitCloner.cloneRepo(repoUrl);
+        public boolean cloneRepo(String repoUrl, String artifact, String group,
```

Comment:
Linked with my previous comment: you could make this method `void` and get rid of `cloned` variable if you delegate catching exceptions to the `consume` method.

Reply:
Since refactoring of error handling will be done in the next version, there are no exceptions to catch, so no need to change this at the moment.